cirq-google/cirq_google/ops/common_gate_families.py

```diff
@@ -0,0 +1,362 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Common Gate Families used in cirq-google"""
+
+from typing import cast, Dict, Optional, Type, Sequence, Union, Any, Callable
+
+import sympy
+import numpy as np
+import cirq
+from cirq.ops.gateset import _gate_str
+
+POSSIBLE_FSIM_GATES = Union[
+    cirq.FSimGate,
+    cirq.PhasedFSimGate,
+    cirq.ISwapPowGate,
+    cirq.PhasedISwapPowGate,
+    cirq.CZPowGate,
+    cirq.IdentityGate,
+]
+
+
+def _exp(theta: Union[complex, sympy.Basic]):
+    """Utility method to return exp(theta) using numpy or sympy, depending on the type of theta."""
+    return sympy.exp(theta) if cirq.is_parameterized(theta) else np.exp(theta)
+
+
+def _gate_list_to_str(gates: Any, gettr: Callable[[Any], str] = _gate_str) -> str:
+    """Converts a list of gates (types/instances) to string by calling gettr (str/repr) on each."""
+    return f'[{",".join([*map(gettr, gates)])}]'
```

Comment:
nit: join can take an iterator, so don't need to capture in a list. I'd suggest using a generator expression instead of calling `map`:
```suggestion
    return f'[{",".join(gettr(g) for g in gates)}]'
```